src/MSBuild.Abstractions/MSBuildWorkspace.cs

```diff
@@ -237,43 +237,33 @@ private bool IsSupportedProjectType(MSBuildProjectRootElement root)
-                        return false;
-                    case ProjectSupportType.UnknownTestProject:
-                        Console.WriteLine($"'{root.FullPath}' has invalid Project Type Guids for test projects and is not supported.");
-                        return false;
-                    case ProjectSupportType.UnsupportedTestType:
-                        Console.WriteLine($"'{root.FullPath}' is an unsupported MSTest test type. Only Unit Tests are supported.");
-                        return false;
-                    case ProjectSupportType.Desktop:
-                    case ProjectSupportType.MSTest:
-                        // Let them know about System.Web while we're here
-                        if (MSBuildHelpers.IsProjectReferencingSystemWeb(root))
-                        {
-                            Console.WriteLine($"'{root.FullPath}' references System.Web, which is unsupported on .NET Core. You may have significant work remaining after conversion.");
-                        }
-                        return true;
-                    case ProjectSupportType.Unsupported:
-                    default:
-                        var allSupportedProjectTypeGuids = DesktopFacts.KnownSupportedDesktopProjectTypeGuids.Select(ptg => ptg.ToString());
-                        var allReadProjectTypeGuids = MSBuildHelpers.GetAllProjectTypeGuids(root);
-                        Console.WriteLine($"{root.FullPath} is an unsupported project type. Not all project type guids are supported.");
-                        PrintGuidMessage(allSupportedProjectTypeGuids, allReadProjectTypeGuids);
-                        return false;
+                case ProjectSupportType.LegacyWeb:
+                    Console.WriteLine($"'{root.FullPath}' is a legacy web project and/or reference System.Web. Legacy Web projects and System.Web are unsupported on .NET Core. You will need to rewrite your application or find a way to not depend on System.Web to convert this project.");
+                    return false;
+                case ProjectSupportType.CodedUITest:
+                    Console.WriteLine($"'{root.FullPath}' is a coded UI test. Coded UI tests are deprecated and not convertable to .NET Core.");
+                    return false;
+                case ProjectSupportType.UnknownTestProject:
+                    Console.WriteLine($"'{root.FullPath}' has invalid Project Type Guids for test projects and is not supported.");
+                    return false;
+                case ProjectSupportType.UnsupportedTestType:
+                    Console.WriteLine($"'{root.FullPath}' is an unsupported MSTest test type. Only Unit Tests are supported.");
+                    return false;
+                case ProjectSupportType.Desktop:
+                case ProjectSupportType.MSTest:
+                    return true;
+                case ProjectSupportType.Unsupported:
+                default:
```

Comment:
I suppose this could be adjusted to:

* Check if there are project type guids
* If they exist, fail
* If they don't, assume it's supported

Reply:
Yeah I'll just do that